src/components/SignUp/index.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState } from "react";
+import { Row, Form, Col, Button, Container, Alert } from "react-bootstrap";
+import { Redirect } from "react-router-dom";
+import { auth } from "../../firebaseConfig";
+import "./index.css";
+import { useTranslation } from "react-i18next";
+
+const SignUp = () => {
+  const { t } = useTranslation();
+  const [redirect, setRedirect] = useState(false);
+  const [alert, setAlert] = useState(null);
+  const [signupInfo, setSignupInfo] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    repeatPassword: "",
+  });
+
+  const handleSignup = async (e) => {
+    e.preventDefault();
+    if (signupInfo.password !== signupInfo.repeatPassword) {
+      setAlert({ message: t("signup.alertMsg"), status: "danger" });
+      return;
+    }
+    try {
+      const response = await auth.createUserWithEmailAndPassword(
+        signupInfo.email,
+        signupInfo.password
+      );
+      response.user.sendEmailVerification();
+      response.user.updateProfile({
+        displayName: signupInfo.firstName + " " + signupInfo.lastName,
+      });
+    } catch (error) {
+      setAlert({ message: error.message, status: "danger" });
+      return;
+    }
+    setSignupInfo({
+      firstName: "",
+      lastName: "",
+      email: "",
+      password: "",
+      repeatPassword: "",
+    });
+    setAlert({ message: t("signup.alertCmplt"), status: "success" });
+    setTimeout(() => setRedirect(true), 1000);
+  };
+
+  const handleChange = (key) => (e) => {
+    setSignupInfo({ ...signupInfo, [key]: e.target.value });
+  };
+
+  return (
+    <Container className="signupContainer">
+      <Row className="webLogo">
+        <h2>No Barriers</h2>
+      </Row>
+      <Row>
+        <h4 className="welcometext">{t("signup.welcomeMsg")}</h4>
+      </Row>
+      <Form onSubmit={handleSignup} className="form">
+        <Form.Row>
+          <Form.Group as={Col} controlId="formGridFirstName" className="mt-3">
+            <Form.Control
+              required
+              type="name"
+              placeholder={t("signup.firstName")}
+              value={signupInfo.firstName}
+              onChange={handleChange("firstName")}
```

Comment:
Nice, so we don't need to pass the event directly, do we?

Reply:
yes we don't.